gbpservice/neutron/nsf/configurator/agents/vpn.py

```diff
@@ -0,0 +1,617 @@
+    """
+    RPC_API_VERSION = '1.0'
+    target = messaging.Target(version=RPC_API_VERSION)
+
+    def __init__(self, conf, sc):
+        self.conf = conf
+        self._sc = sc
+
+    def vpnservice_updated(self, context, **kwargs):
+        arg_dict = {'context': context,
+                    'kwargs': kwargs}
+
+        ev = self._sc.event(id='UPDATE_VPN_SERVICE', data=arg_dict)
+        self._sc.rpc_event(ev)
+
+
+class VPNaasEventHandler(object):
+    """
+    Handler class for demultiplexing vpn configuration
+    requests from VPNaas Plugin and sending to appropriate driver.
+    """
+    def __init__(self, sc, drivers):
+        self._sc = sc
+        self.drivers = drivers
+        self.needs_sync = True
+        self.context = context.get_admin_context_without_session()
+        self.plugin_rpc = VpnaasRpcSender(
+            const.VPN_PLUGIN_TOPIC,
+            self.context)
+
+    def _get_driver(self, data):
+        ''' TO DO: Do demultiplexing logic based on vendor
+        '''
+        svc_type = data.get('kwargs').get('svc_type')
+        if svc_type == const.SERVICE_TYPE_IPSEC:
+            self.ipsec_driver = self.drivers["vyos_ipsec_vpnaas"]
+            return self.ipsec_driver
+        elif svc_type == const.SERVICE_TYPE_OPENVPN:
+            self.ssl_driver = self.drivers["vyos_ssl_vpnaas"]
+            return self.ssl_driver
+
+    def handle_event(self, ev):
+        try:
+            msg = ("Worker process with ID: %s starting "
+                   "to handle task: %s of topic: %s. "
+                   % (os.getpid(), ev.id, const.VPN_GENERIC_CONFIG_RPC_TOPIC))
+            LOG.debug(msg)
+
+            driver = self._get_driver(ev.data)
+            method = getattr(driver, "%s" % (ev.id.lower()))
+            method(ev)
+        except Exception as err:
+            LOG.error("Failed to perform the operation: %s. %s"
+                      % (ev.id, str(err).capitalize()))
+        finally:
+            self._sc.event_done(ev)
+
+    def vpnservice_updated(self, ev, driver):
+        context = ev.data.get('context')
+        kwargs = ev.data.get('kwargs')
+        LOG.debug(_("Vpn service updated from server side"))
+
+        try:
+            driver.vpnservice_updated(context,  **kwargs)
+        except Exception as err:
+            LOG.error("Failed to update VPN service. %s"
+                      % str(err).capitalize())
+
+        reason = kwargs.get('reason')
+        rsrc = kwargs.get('rsrc_type')
+
+        if (reason == 'delete' and rsrc == 'ipsec_site_connection'):
+            conn = kwargs['resource']
+            resource_id = conn['id']
+            self.plugin_rpc.ipsec_site_conn_deleted(context,
+                                                    resource_id=resource_id)
+        elif (reason == 'delete' and rsrc == 'ssl_vpn_connection'):
+            conn = kwargs['resource']
+            resource_id = conn['id']
+            self.plugin_rpc.ssl_vpn_conn_deleted(context,
```

Comment:
Remove this SSL connection deleted function. Also, check for any other references to SSL in the code and remove.